From 7882234c2f43f006409fcce22e61a67ce171181b Mon Sep 17 00:00:00 2001
From: Ammar Faizi <ammarfaizi2@gmail.com>
Date: Sat, 11 Dec 2021 10:06:29 +0700
Subject: [PATCH 1/5] render_frame_host_impl.h: Make `bool
 CanExecuteJavaScript()` be public

We will need to call this method from public scope to make sure we can
execute JavaScript for preloading. So, make it public.

No functional changes are intended for this patch.

Signed-off-by: Ammar Faizi <ammarfaizi2@gmail.com>
---
 content/browser/renderer_host/render_frame_host_impl.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/content/browser/renderer_host/render_frame_host_impl.h b/content/browser/renderer_host/render_frame_host_impl.h
index a93bb313bc..0430290154 100644
--- a/content/browser/renderer_host/render_frame_host_impl.h
+++ b/content/browser/renderer_host/render_frame_host_impl.h
@@ -309,6 +309,11 @@ class CONTENT_EXPORT RenderFrameHostImpl
 
   ~RenderFrameHostImpl() override;
 
+  // Returns true if the ExecuteJavaScript() API can be used on this host.
+  // The checks do not apply to ExecuteJavaScriptInIsolatedWorld, nor to
+  // ExecuteJavaScriptForTests.  See also AssertNonSpeculativeFrame method.
+  bool CanExecuteJavaScript();
+
   // RenderFrameHost
   int GetRoutingID() override;
   const blink::LocalFrameToken& GetFrameToken() override;
@@ -2828,11 +2833,6 @@ class CONTENT_EXPORT RenderFrameHostImpl
       bool* disable_secure_dns = nullptr,
       network::mojom::URLLoaderFactoryOverridePtr* factory_override = nullptr);
 
-  // Returns true if the ExecuteJavaScript() API can be used on this host.
-  // The checks do not apply to ExecuteJavaScriptInIsolatedWorld, nor to
-  // ExecuteJavaScriptForTests.  See also AssertNonSpeculativeFrame method.
-  bool CanExecuteJavaScript();
-
   // Returns the AXTreeID of the parent when the current frame is a child frame
   // (i.e. not a main frame) or when it's an embedded browser plugin guest, or
   // ui::AXTreeIDUnknown() otherwise.
-- 
2.32.0

